internal/core/thirdparty/google_cloud_sdk/CMakeLists.txt

```diff
@@ -0,0 +1,142 @@
+        GIT_REPOSITORY https://github.com/nlohmann/json.git
+        GIT_TAG ${GOOGLE_JSON_VERSION}
+        PREFIX ${CMAKE_BINARY_DIR}/3rdparty_download/google-json-subbuild
+        BUILD_COMMAND ${GOOGLE_JSON_BUILD_COMMAND}
+        INSTALL_COMMAND ${GOOGLE_JSON_INSTALL_COMMAND}
+        CMAKE_ARGS ${GOOGLE_JSON_CMAKE_ARGS}
+    )
+
+    message(STATUS "Building GOOGLE_SDK-${GOOGLE_SDK_VERSION} from source")
+
+    set(GOOGLE_SDK_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX})
+
+    set(GOOGLE_SDK_BUILD_COMMAND make)
+    set(GOOGLE_SDK_INSTALL_COMMAND make install)
+
+    set(GOOGLE_SDK_CMAKE_ARGS
+        "-DCMAKE_BUILD_TYPE=Release"
+        "-DBUILD_DEPS=OFF"
+        "-DBUILD_TESTING=OFF"
+        "-DBUILD_SHARED_LIBS=ON"
+        "-DGOOGLE_CLOUD_CPP_ENABLE=storage"
+        "-DGOOGLE_CLOUD_CPP_ENABLE_EXAMPLES=OFF"
+        "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}")
+
+    ExternalProject_Add(google_sdk_ep
+        GIT_REPOSITORY https://github.com/googleapis/google-cloud-cpp.git
+        GIT_TAG ${GOOGLE_SDK_VERSION}
+        PREFIX ${CMAKE_BINARY_DIR}/3rdparty_download/google-sdk-subbuild
+        BUILD_IN_SOURCE 1
+        BUILD_COMMAND ${GOOGLE_SDK_BUILD_COMMAND}
+        INSTALL_COMMAND ${GOOGLE_SDK_INSTALL_COMMAND}
+        CMAKE_ARGS ${GOOGLE_SDK_CMAKE_ARGS}
+    )
+    add_dependencies(google_sdk_ep google_abseil_ep)
+    add_dependencies(google_sdk_ep google_json_ep)
+    add_dependencies(google_sdk_ep google_crc32_ep)
+
+    add_library(google_cloud_cpp_storage SHARED IMPORTED)
+    set_target_properties(google_cloud_cpp_storage
+        PROPERTIES
+        IMPORTED_GLOBAL TRUE
+        IMPORTED_LOCATION ${GOOGLE_SDK_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/${CMAKE_SHARED_LIBRARY_PREFIX}google_cloud_cpp_storage${CMAKE_SHARED_LIBRARY_SUFFIX}
+        INTERFACE_INCLUDE_DIRECTORIES ${GOOGLE_SDK_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR})
+    add_dependencies(google_cloud_cpp_storage google_sdk_ep)
+
+    add_library(google_cloud_cpp_common SHARED IMPORTED)
+    set_target_properties(google_cloud_cpp_common
+        PROPERTIES
+        IMPORTED_GLOBAL TRUE
+        IMPORTED_LOCATION ${GOOGLE_SDK_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/${CMAKE_SHARED_LIBRARY_PREFIX}google_cloud_cpp_common${CMAKE_SHARED_LIBRARY_SUFFIX}
+        INTERFACE_INCLUDE_DIRECTORIES ${GOOGLE_SDK_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR})
+    add_dependencies(google_cloud_cpp_common google_sdk_ep)
+
+    add_library(google_cloud_cpp_rest_internal SHARED IMPORTED)
+    set_target_properties(google_cloud_cpp_rest_internal
+        PROPERTIES
+        IMPORTED_GLOBAL TRUE
+        IMPORTED_LOCATION ${GOOGLE_SDK_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/${CMAKE_SHARED_LIBRARY_PREFIX}google_cloud_cpp_rest_internal${CMAKE_SHARED_LIBRARY_SUFFIX}
+        INTERFACE_INCLUDE_DIRECTORIES ${GOOGLE_SDK_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR})
+    add_dependencies(google_cloud_cpp_rest_internal google_sdk_ep)
+endmacro()
+
+build_google_sdk_s3()
```

Comment:
Could the imported gcp be compiled on mac？

Reply:
not valid it now, just support it at linux 
+# ******************************* Thirdparty google cloud sdk ********************************
+if ( LINUX AND BUILD_DISK_ANN STREQUAL "ON" )
+    add_subdirectory( google_cloud_sdk )
+endif()
+